packages/core/src/utils/security/handlers/apiKey.ts

```diff
@@ -0,0 +1,38 @@
+import { fromNullable, getOrElse, mapNullable } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+import { SecurityScheme } from './types';
+import { when } from './utils';
+
+export const apiKeyInCookie = {
+  test: ({ type, in: where }: SecurityScheme) => where === 'cookie' && type === 'apiKey',
+  handle: <R, I>(someInput: I, name: string, resource?: R) => {
+    const probablyCookie = get(someInput, ['headers', 'cookie']);
+
+    const isApiKeyInCookie = pipe(
+      fromNullable(probablyCookie),
+      mapNullable(cookie => new RegExp(`${name}=.+`).test(cookie)),
+      getOrElse(() => false),
+    );
+
+    return when<R>(isApiKeyInCookie, '', resource);
+  },
+};
+
+export const apiKeyInHeader = {
+  test: ({ type, in: where }: SecurityScheme) => where === 'header' && type === 'apiKey',
+  handle: <R, I>(someInput: I, name: string, resource?: R) => {
+    const isAPIKeyProvided = get(someInput, ['headers', name.toLowerCase()]);
```

Comment:
What was the reason of making those functions generic (<R, I> stuff)?
More specific types could benefit from static type checking and you could avoid `get(...)`

Reply:
@chris-miaskowski, it's just that `factory` function is generic and I had no choice (?)